CMakeLists.txt

```diff
@@ -94,20 +66,24 @@ if (GTest_FOUND)
 endif()
 
 option(BUILD_EXAMPLES "Build examples" ON)
-if (BUILD_EXAMPLES)
+if (BUILD_MASTER AND BUILD_EXAMPLES)
```

Comment:
Proposal for the cleanup not sure if it is necessarily better but here is a slight rework for the CMakeLists to build:
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0031c79..b76c6ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,25 +65,12 @@ if (GTest_FOUND)
   endif()
 endif()
 
-option(BUILD_EXAMPLES "Build examples" ON)
-if (BUILD_MASTER AND BUILD_EXAMPLES)
-  if (UNIX)
-      add_subdirectory(examples/master)
-  endif()
-endif()
-
-if (BUILD_SLAVE AND BUILD_EXAMPLES)
-  add_subdirectory(examples/slave)
-endif()
-
-option(BUILD_SIMULATION "Build simulation" ON)
-if (BUILD_MASTER AND BUILD_SIMULATION)
-  if (UNIX)
-      add_subdirectory(simulation)
-  endif()
-endif()
-
-option(BUILD_TOOLS "Build tools" ON)
-if (BUILD_MASTER AND BUILD_TOOLS)
+if (BUILD_MASTER)
+  add_subdirectory(simulation)
   add_subdirectory(tools)
 endif()
+
+option(BUILD_EXAMPLES "Build examples" ON)
+if (BUILD_EXAMPLES)
+    add_subdirectory(examples)
+endif()
\ No newline at end of file
diff --git a/simulation/CMakeLists.txt b/simulation/CMakeLists.txt
index f611a5a..8504cda 100644
--- a/simulation/CMakeLists.txt
+++ b/simulation/CMakeLists.txt
@@ -1,3 +1,8 @@
-add_executable(network_simulator network_simulator.cc ESC.cc)
-set_kickcat_properties(network_simulator)
-target_link_libraries(network_simulator kickcat)
+option(BUILD_SIMULATION "Build simulation" ON)
+if (BUILD_SIMULATION)
+    if (UNIX)
+        add_executable(network_simulator network_simulator.cc ESC.cc)
+        set_kickcat_properties(network_simulator)
+        target_link_libraries(network_simulator kickcat)
+    endif()
+endif()
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index b1eddfb..3de18fc 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,17 +1,21 @@
-add_executable(scanTopology scanTopology.cc)
-target_link_libraries(scanTopology kickcat)
-set_kickcat_properties(scanTopology)
+option(BUILD_TOOLS "Build tools" ON)
 
-add_executable(debugUseCase debugUseCase.cc)
-target_link_libraries(debugUseCase kickcat)
-set_kickcat_properties(debugUseCase)
+if (BUILD_TOOLS)
+    add_executable(scanTopology scanTopology.cc)
+    target_link_libraries(scanTopology kickcat)
+    set_kickcat_properties(scanTopology)
 
-add_executable(eeprom eeprom.cc)
-target_link_libraries(eeprom kickcat)
-set_kickcat_properties(eeprom)
+    add_executable(debugUseCase debugUseCase.cc)
+    target_link_libraries(debugUseCase kickcat)
+    set_kickcat_properties(debugUseCase)
 
-if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    add_executable(check_network_stability checkNetworkStability.cc)
-    target_link_libraries(check_network_stability kickcat)
-    set_kickcat_properties(check_network_stability)
+    add_executable(eeprom eeprom.cc)
+    target_link_libraries(eeprom kickcat)
+    set_kickcat_properties(eeprom)
+
+    if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+        add_executable(check_network_stability checkNetworkStability.cc)
+        target_link_libraries(check_network_stability kickcat)
+        set_kickcat_properties(check_network_stability)
+    endif()
 endif()
```

Reply:
So the whole part was changed by using the if(NUTTX) way